Changed handlers

GrayDog0 opened the pull request:
Changed top_ handlers, added unique comments to the display for the first ones in the list

GrayDog0 committed:
test

denis1011101 commented:
мужики дисклеймер! хочется сильно не душнить и шоб была творческая атмосфера, так шо любые предложения это бомба, удачи! ✨

denis1011101 commented:
Серёг давай переназовём имя пр'а и имя коммита, например `feature(tops): decoration tops`, либо что-то подобное, тип сперва `feature` - если чёт новое завозим, `fix` - если правим что-то, `chore` - если правим конфиги или что-то подобное что не является ни фичей ни фиксом, потом в скобках область которую трогаем - имя фичи либо имя самого файла и потом краткое описание правок, но это ток пример, есть куча других практик, главное шоб понятно было

denis1011101 commented on app/handlers/top_giga_handler.go:
```diff
@@ -6,17 +6,22 @@ import (
 	"log"
 	"strings"
 
-	"github.com/denis1011101/super_cm_bot/app"
```
Серёг давай вернём импорт, потому что ниже будем юзать функцию оттуда `app.SendMessage`

denis1011101 commented on app/handlers/top_unhandsome_handler.go:
```diff
@@ -36,19 +41,39 @@ func TopUnhandsome(update tgbotapi.Update, bot *tgbotapi.BotAPI, db *sql.DB) {
+	message := sb.String()
+
 	// Отправка сообщения
-	app.SendMessage(chatID, sb.String(), bot, update.Message.MessageID)
```
тут ты написал функцию заново, а лучше юзать [мою](https://github.com/GrayDog0/super_cm_bot/blob/main/app/bot.go#L10), это лаконичнее и приятнее)

GrayDog0 replied:
Я написал заного из-за того, что не выходило по какой-то причине отправлять уникальные комменты с твоей функцией.

Ок, завтра заменю функцию и верну импорт "github.com/denis1011101/super_cm_bot/app"

Keklil commented on app/handlers/top_giga_handler.go:
```diff
@@ -6,17 +6,22 @@ import (
 )
 
+type TopGigaStruct struct {
+	ID      int    `db:"handsome_count"`
+	Data    string `db:"pen_name"`
```
Может подправить нейминг? Назначение с названием не сходится.
Зачем здесь теги?

Keklil replied:
Оо, это видимо отсылка к либе sqlx. Можем заюзать ее, тогда эти теги пригодятся

Вместо нашего цикла с rows.Next() будет такая крастота:
```
rows, err := sqlx.Select(db, &records, `
	SELECT pen_length, pen_name
	FROM pens 
	WHERE tg_chat_id = ? 
	ORDER BY pen_length DESC 
	LIMIT 10
`, chatID)
```

GrayDog0 replied:
@Keklil 

Изначально я учился как привязать значения из структры к запросу, но по итогу забил и просто поменял в запросе местами получаемые данные.
Добавить sqlx?

Keklil replied:
Давай с sqlx попробуем, да
Либа уже заимпорчена даже, кажется

denis1011101 replied:
мужики я тут недавно сделал `go mod tidy` и неиспользуемые зависимости потёрлись, так шо надо будет снова добавить [вот тут](https://github.com/denis1011101/super_cm_bot/commit/43c6ad706f855b07c49da5c27d265943535d4281#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6L15)

denis1011101 commented on app/handlers/top_unhandsome_handler.go:
```diff
@@ -36,19 +41,39 @@ func TopUnhandsome(update tgbotapi.Update, bot *tgbotapi.BotAPI, db *sql.DB) {
 	defer rows.Close()
 
+	var records []TopUnhandsomeStruct
+	uniqueComments := []string{"Самый крепкий анус на деревне 🐓", "Около пидорства 💩"}
```
Серёг там было больше словечек давай добавим остальные или если будут идеи какие-то свои словечки, не ток в этом топе, во всех

GrayDog0 replied:
Я вроде вставил все слова из кода на пыхе, если есть идеи - пиши, добавлю.
Не знаешь, кстати какую-нибудь нейросетку годную, чтоб она придумала?
Можно хоть на каждого из списка отдельный комент записать

denis1011101 replied:
идей пока нет, ща напишу какие ещё были

denis1011101 replied:
а вроде рил все

denis1011101 commented:
такой 🤡, это Саня про себя ес шо, бывает!

Keklil commented:
Денис, не бань меня за это ревью, я реально клоун

denis1011101 commented on app/handlers/top_length_handler.go:
```diff
@@ -35,19 +42,43 @@ func TopLength(update tgbotapi.Update, bot *tgbotapi.BotAPI, db *sql.DB) {
 	defer rows.Close()
 
+	var records []TopLengthStruct
+	uniqueComments := []string{"Настоящий гигачад с елдой ", "Полупокер но с большим хреном ", "Лучше быть третьим чем выступать в цирке "}
```
во забыли одного
```php
            if ($idx === (count($penises) - 1)) {
                $text = '%s. @%s куколд с %s см 🤡';
            }
```

denis1011101 commented on app/handlers/top_length_handler.go:
```diff
@@ -35,19 +42,43 @@ func TopLength(update tgbotapi.Update, bot *tgbotapi.BotAPI, db *sql.DB) {
+	uniqueComments := []string{"Настоящий гигачад с елдой ", "Полупокер но с большим хреном ", "Лучше быть третьим чем выступать в цирке "}
+	commonComment := "У него писунька "
+	uniqueComments1 := []string{" см 😱", " см 💪", " см 🐺"}
+	commonComment1 := " см 🤡"
```
считаю что имена переменных тут как фишка, шоб интереснее было код читать в будущем 😜

GrayDog0 committed:
Merge branch 'denis1011101:main' into main

GrayDog0 committed:
feature(tops): decoration tops

Changed top_ handlers, added unique comments to the display for the first ones in the list
